tasks/main.yml

```diff
@@ -90,32 +100,33 @@
         loop_var: user
         label: "{{ user.name }} {{ user.state }}"
       when:
-        # Ansible user module fails to modify any logged in user when uid is set
-        # Workaround is to run usermod manually for ansible_ssh_user
-        - user.uid is not defined or (user.uid is defined and user.name != ansible_ssh_user)
+        # Ansible user module and usermod command fail to modify any logged in user when uid is set
+        # Workaround is to run usermod manually for logged in users and not set UID even if it is defined
+        - user.uid is not defined or (user.uid is defined and user.name not in admin_logged_in_users)
```

Comment:
Daaaaamn!

Reply:
Definitely ;) 
Added change_when here and there and updated some more documentation/styling while at it.
 (#31 was an old PR that hung for a while and eventually conflicted with some other updates. )